src/DurableTask.SqlServer.AzureFunctions/SqlDurabilityProvider.cs

```diff
@@ -200,13 +197,12 @@ public override bool TryGetScaleMonitor(
             string storageConnectionString,
             out IScaleMonitor scaleMonitor)
         {
-            if (this.scaleMonitor == null)
+            if (this.singletonSqlMetricsProvider == null)
             {
-                var sqlMetricsProvider = new SqlMetricsProvider(this.service);
-                this.scaleMonitor = new SqlScaleMonitor(hubName, sqlMetricsProvider);
+                this.singletonSqlMetricsProvider = new SqlMetricsProvider(this.service);
```

Comment:
One last thing. I recall the final fix that Alexey made to the Azure Storage metrics provider was to throttle it so that we only fetch metrics once every few seconds or so. This is how we ensured that we don't have too many storage transactions, even if the app has a lot of triggers. I think it will be important to do that for the `SqlMetricsProvider` too so that we don't overwhelm the customer's database with expensive SQL queries.